### DISCUSSION TITLE
Continuous zoom

### DISCUSSION THREAD

**Discussion by diehl** [opengeos/leafmap#154] [Q&A] [answered]
When using the default backend, is it possible to realize a continuous zoom experience similar to that offered by kepler.gl? Would using Mapbox tiles with the default backend deliver that experience? I'm seeing the default backend step in discrete zoom levels as I zoom into the map which is what I'd like to avoid if possible.

**Comment by giswqs** [comment, marked as answer]
You can set the `zoom_snap` parameter (`Forces the map’s zoom level to always be a multiple of this`) when creating the map. See https://ipyleaflet.readthedocs.io/en/latest/api_reference/map.html



```python
m = leafmap.Map(zoom_snap=0.1)
```

**Reply by diehl** [reply]
This does improve the transition - thank @giswqs - and yet, it's not as responsive as Kepler. Do you have an example showing how to use Mapbox tiles with the default backend? From the documentation, it looks like that might be possible.

**Reply by giswqs** [reply]
That needs to be resolved by ipyleaflet. There is not much I can do about it. Since leafmap includes the Kelpler backend, why not just use it?

```python
import leafmap.kepler as leafmap
m = leafmap.Map()
M
```

**Reply by diehl** [reply]
I was wondering if I'd lose implementation options that are offered by the other backends by using kepler. It seems like there's more functionality available with the default backend. Maybe I'm wrong?

I see a reference to the Mapbox key for using Mapbox tiles here: https://leafmap.org/foliumap/?h=mapbox#leafmap.foliumap.Map.add_tile_layer Wondering if that allows me to get the best of both worlds.

**Reply by giswqs** [reply]
You are correct. The default ipyleaflet plotting backend has the richest functionality. It is possible to add Mapbox tiles. See the example below. 

https://xyzservices.readthedocs.io/en/stable/api.html

```python
import leafmap
import xyzservices.providers as xyz
tile_url = xyz.MapBox.build_url(accessToken="my_token")
m = leafmap.Map()
m.add_tile_layer(tile_url, name, attribution)

```

**Reply by diehl** [reply]
@giswqs brilliant - thank you for this example! I'll give this a try.

**Reply by giswqs** [reply]
I just added the plotly backend. You will soon be able to use any Mapbox tiles. https://github.com/giswqs/leafmap/pull/155

**Reply by diehl** [reply]
Even better!
